### PR TITLE
More gradle lint

### PR DESCRIPTION
**Pull request by mikehardy**
## Purpose / Description

Much to my annoyance I continue to see gradle 9 deprecation warnings in Android Studio

I hunted down a couple more with generic "./gradlew test" or "./gradlew build" type targets

None of this is really important.

## Fixes

Nothing logged, but I don't want unexpected build errors when gradle 9 comes out, and it should be shortly

## Approach

The gradle deprecation warnings are at least very informative, they contain links to the page describing what to do

## How Has This Been Tested?

These changes were basically all test-related, so, testing was pretty obviously to keep running the test targets

## Learning (optional, can help others)

just handling deprecation in toolchains is a significant chunk of my life

Also, it seems like there might be some internal chunk of Android Studio that contains deprecation itself, as one final target seems to continue triggering things - `amazonDebugRuntimeClasspathCopy` but I can't locate that target so I dunno

We may find out when gradle 9 launches

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -76,7 +76,8 @@ jetbrainsAnnotations = "26.0.1"
 json = "20240303"
 jsoup = "1.18.1"
 androidTestJunit = "1.2.1"
-junitJupiter= "5.11.3"
+junitJupiter = "5.11.3"
```
Comment (mikehardy) [inline gradle/libs.versions.toml:79]: one spacing change, I just threw it in there. Forgive me

**Review by david-allison — APPROVED** [review]
Looks great!

Does this fix:

* https://github.com/ankidroid/Anki-Android/issues/16618

File: testlib/src/test/java/com/ichi2/testutils/common/AnkiAssertTest.kt [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            assertThrows<IllegalArgumentException>("IllegalArgumentException is not found") {
+                throw NullPointerException("this is unexpected")
+            }
+        } catch (e: AssertionError) {
+            // we expect this
+            assertThat(
+                "message is expected",
+                e.message?.contains("unexpected exception type thrown") == true
+            )
+            return
```
Comment (Arthur-Milchior) [inline testlib/src/test/java/com/ichi2/testutils/common/AnkiAssertTest.kt:31]: I admit it took me a minute to understand the expected route through the code.
I'd consider maybe,
```kotlin
    @Test
    fun failsWithIncorrectException() {
        var assertionErrorThrown = false
        try {
            assertThrows<IllegalArgumentException>("IllegalArgumentException is not found") {
                throw NullPointerException("this is unexpected")
            }
        } catch (e: AssertionError) {
            // we expect this
            assertThat(
                "message is expected",
                e.message?.contains("unexpected exception type thrown") == true
            )
            assertionErrorThrown = true
    }
    assertThat("Should have had an AssertionError", assertionErrorThrown)
}
```

This way, I don't have to think about why there is an inconditional fail at the end of the test

**Review by Arthur-Milchior — APPROVED** [review]

File: testlib/src/test/java/com/ichi2/testutils/common/AnkiAssertTest.kt (hunk 2 above)
Comment (mikehardy) [reply testlib/src/test/java/com/ichi2/testutils/common/AnkiAssertTest.kt:31]: 🤔 I think it's still cleaner without an extra local var and comparison but if it is unclear at all I think there is an opportunity to improve the message in the fail itself so it's completely clear. What if it was like

`fail("only reachable if no exception or exception did match correctly - both are always a failure")`

**Comment by mikehardy** [comment]
> Looks great!
> 
> Does this fix:
> 
> * [CI: `common` module: tests don't run #16618](https://github.com/ankidroid/Anki-Android/issues/16618)

I did not test that but there is no real rush getting this in. I'll give it a look and see

File: testlib/src/test/java/com/ichi2/testutils/common/AnkiAssertTest.kt (hunk 2 above)
Comment (Arthur-Milchior) [reply testlib/src/test/java/com/ichi2/testutils/common/AnkiAssertTest.kt:31]: That's a weak opinion, if you prefer this version, fine by me

**Comment by david-allison** [comment]
Let's see if it fixes the `common` testing issue 🤞

* https://github.com/david-allison/Anki-Android/actions/runs/11786385567

**Comment by david-allison** [comment]
Adding a https://github.com/ankidroid/Anki-Android/labels/Needs%20Author%20Reply on the test changes (or not): https://github.com/ankidroid/Anki-Android/pull/17363/files#r1836879056

I'm ambivalent. Implementer's choice, as long as CI is green, happy to merge

File: testlib/src/test/java/com/ichi2/testutils/common/AnkiAssertTest.kt (hunk 2 above)
Comment (mikehardy) [reply testlib/src/test/java/com/ichi2/testutils/common/AnkiAssertTest.kt:31]: I'm going to guess (correct if I am wrong!) that the weakness was between my take on it and Arthur's which were both the same basic implementation. Yours is a totally different thing and appears more expressive to me, I'll adopt it

**Comment by mikehardy** [comment]
I adopted the test kotlinization you proposed David
Also noticed our lint wasn't applied in testlib or common so added a comment where I apply our lint config

**Comment by mikehardy** [comment]
Added a (final?) commit that disables the custom scheduler test as Flaky in CI again. Sad to do it but it's causing a lot of issues

File: testlib/src/test/java/com/ichi2/testutils/common/AnkiAssertTest.kt [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import org.hamcrest.Matchers.containsString
+import org.junit.Test
+import org.junit.jupiter.api.fail
+
+/**
+ * Tests for methods in [assertThrows]
+ */
+class AnkiAssertTest {
+
+    @Test fun failsWithIncorrectException() {
```
Comment (david-allison) [inline testlib/src/test/java/com/ichi2/testutils/common/AnkiAssertTest.kt]: Unusual: `@Test` is normally on the line above the `fun`

**Review by david-allison — APPROVED** [review]
Force pushed a very minor nit. Let's go!

**Pull request merged**